presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastoreClient.java

```diff
@@ -44,18 +46,28 @@
 import java.util.List;
 import java.util.Map;
 
+import static com.google.common.reflect.Reflection.newProxy;
 import static java.util.Objects.requireNonNull;
 
 public class ThriftHiveMetastoreClient
         implements ThriftMetastoreClient
 {
+    private static final Logger log = Logger.get(ThriftHiveMetastoreClient.class);
+
     private final TTransport transport;
-    private final ThriftHiveMetastore.Client client;
+    private final ThriftHiveMetastore.Iface client;
 
     public ThriftHiveMetastoreClient(TTransport transport)
     {
         this.transport = requireNonNull(transport, "transport is null");
-        this.client = new ThriftHiveMetastore.Client(new TBinaryProtocol(transport));
+        ThriftHiveMetastore.Client client = new ThriftHiveMetastore.Client(new TBinaryProtocol(transport));
+        if (log.isDebugEnabled()) {
```

Comment:
Looking this up once means that we can't enable or disable it via JMX. I don't think there's any downside to doing the check at runtime.

Reply:
`ThriftHiveMetastoreClient` is short-lived